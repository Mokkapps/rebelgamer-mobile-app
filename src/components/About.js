// @flow

import { Button, Image, Linking, Platform, StyleSheet, View } from 'react-native';
import React from 'react';
import { Text } from 'react-native-elements';
import email from 'react-native-email';

import { version } from './../../package.json';
import DeviceDetector from './../utils/DeviceDetector';
import { REBELGAMER_RED, MOKKAPPS_MAIL, REBELGAMER_MAIL, APPLE_APP_ID, GOOGLE_PACKAGE_NAME } from '../constants';
import translate from '../utils/translate';

const tvImage = require('../../assets/tv.png');

const styles = StyleSheet.create({
  appName: {
    fontWeight: 'bold',
    color: 'black'
  },
  container: {
    margin: 10,
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  image: {
    height: 100,
    width: 100,
    resizeMode: 'contain'
  },
  description: {
    marginTop: 20,
    textAlign: 'center',
    color: 'black',
    fontSize: DeviceDetector.isTablet() ? 20 : 15
  },
  button: {
    marginTop: 20,
    width: 200
  }
});

type Props = {};
type State = {};

class About extends React.Component<Props, State> {
  static navigationOptions = {
    title: '',
    headerTintColor: REBELGAMER_RED
  };

  onPressContact = () => {
    const os = Platform.OS === 'ios' ? 'iOS' : 'Android';
    const to = [MOKKAPPS_MAIL, REBELGAMER_MAIL];
    email(to, { subject: `${translate('mailSubject')} (Version: ${version}, OS: ${os})` }).catch(console.error);
  };

  onPressRate = async () => {
    if (Platform.OS === 'ios') {
      await Linking.openURL(`itms-apps://itunes.apple.com/app/id${APPLE_APP_ID}`);
    } else {
      await Linking.openURL(`http://play.google.com/store/apps/details?id=${GOOGLE_PACKAGE_NAME}`);
    }
  };

  render() {
    return (
      <View style={styles.container}>
        <Image style={styles.image} source={tvImage} />
        <Text h3 style={styles.appName}>
          {translate('APP_NAME')}
        </Text>
        <Text>{`${translate('VERSION')} ${version}`}</Text>
        <Text style={styles.description}>{translate('APP_DESCRIPTION')}</Text>
        <View style={styles.button}>
          <Button title={translate('CONTACT_US')} color={REBELGAMER_RED} onPress={this.onPressContact} />
        </View>
        <View style={styles.button}>
          <Button title={translate('RATE_APP')} color={REBELGAMER_RED} onPress={this.onPressRate} />
        </View>
      </View>
    );
  }
}

export default About;
