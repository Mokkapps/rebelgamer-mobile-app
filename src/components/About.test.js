import React from 'react';
import renderer from 'react-test-renderer';

import About from './About';

it('renders correctly', () => {
  const tree = renderer.create(<About />).toJSON();
  expect(tree).toMatchSnapshot();
});
