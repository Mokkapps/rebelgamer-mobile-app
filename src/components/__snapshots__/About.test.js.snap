// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "margin": 10,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 100,
        "resizeMode": "contain",
        "width": 100,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        Object {
          "fontSize": 35,
        },
        undefined,
        undefined,
        undefined,
        Object {},
        undefined,
        undefined,
        Object {
          "color": "black",
          "fontWeight": "bold",
        },
      ]
    }
  >
    RebelGamer
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Version: 3.0.2
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "color": "black",
          "fontSize": 20,
          "marginTop": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    Mit der RebelGamer App bist du immer auf dem aktuellsten Stand in Sachen Videospiele. Wir liefern dir aktuelle Nachrichten, Tests, Vorschauen, „etwas andere“ Reportagen, Interviews und Berichte zu den Themen Games und Digital Entertainment.
  </Text>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": 200,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#F44336",
              },
            ]
          }
        >
          Kontaktiere uns
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": 200,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#F44336",
              },
            ]
          }
        >
          Bewerte die App
        </Text>
      </View>
    </View>
  </View>
</View>
`;
